app/mailers/volunteer_mailer.rb

```diff
@@ -1,8 +1,14 @@
 class VolunteerMailer < ApplicationMailer
   add_template_helper(ApplicationHelper)
-  def welcome_email(volunteer, volunteer_email)
+  def welcome_email(volunteer, signup_email)
     @volunteer = volunteer
-    @volunteer_email = volunteer_email
-    mail(to: volunteer.contact.primary_email, subject: @volunteer_email.subject)
+    @signup_email = signup_email
+    subject = signup_email.subject
+    subject.gsub! '%<anrede>', t("salutation.#{volunteer.salutation}")
+    subject.gsub! '%<name>', "#{volunteer.contact.first_name} #{volunteer.contact.last_name}"
+    body = signup_email.body
+    body.gsub! '%<anrede>', t("salutation.#{volunteer.salutation}")
+    body.gsub! '%<name>', "#{volunteer.contact.first_name} #{volunteer.contact.last_name}"
```

Comment:
This looks very similar to the code in @casaper's [`ReminderMailingVolunteer`](https://github.com/panter/aoz-003/pull/359/files#diff-be29e4532f1777e302a921922fa682e3) model. You can simplify it a bit by defining the hash of replacements (`template_variables = { 'anrede' => ..., ... }`) and then using string formatting (`subject % template_variables`)

Also we should probably use the same placeholder names, i.e. "Anrede" and "Name" (uppercase)

Reply:
I'd definitly agree to use the same placeholder names as the User will see and insert in the frontend.
So In that case it would be german. And why use gsub, if `%{Placeholder}` does the job so much easier?

Reply:
The Syntax `%{}` is native, so no gsub is needed.